SpatialGDK/Source/SpatialGDKEditor/Private/SchemaGenerator/SpatialGDKEditorSchemaGenerator.cpp

```diff
@@ -1579,11 +1578,42 @@ bool ExtractComponentSetsFromSchemaJson(const FString& SchemaJsonPath, TMap<uint
 	TMap<FString, uint32> ComponentMap;
 	TMap<uint32, TSet<FString>> ComponentRefSetMap;
 
+	TMap<uint32, FString> ComponentIdToDataDefinition;
+	TMap<FString, TArray<uint32>> DataDefinitionToFieldIds;
+
 	for (const auto& FileValue : *SchemaFiles)
 	{
 		const TSharedPtr<FJsonObject>* FileObject;
 		SAFE_TRYGET(FileValue, Object, FileObject);
 
+		const TArray<TSharedPtr<FJsonValue>>* TypesDecl;
+		SAFE_TRYGETFIELD((*FileObject), Array, "types", TypesDecl);
+
+		for (const auto& TypeValue : *TypesDecl)
+		{
+			const TSharedPtr<FJsonObject>* TypeObject;
+			SAFE_TRYGET(TypeValue, Object, TypeObject);
+
+			FString ComponentName;
+			SAFE_TRYGETFIELD((*TypeObject), String, "qualifiedName", ComponentName);
+
+			DataDefinitionToFieldIds.Add(ComponentName);
+
+			const TArray<TSharedPtr<FJsonValue>>* FieldArray;
+			SAFE_TRYGETFIELD((*TypeObject), Array, "fields", FieldArray);
+
+			for (const auto& ArrayValue : *FieldArray)
+			{
+				const TSharedPtr<FJsonObject>* ArrayObject;
+				SAFE_TRYGET(ArrayValue, Object, ArrayObject);
+
+				int32 FieldId;
+				SAFE_TRYGETFIELD((*ArrayObject), Number, "fieldId", FieldId);
+
+				DataDefinitionToFieldIds[ComponentName].Add(FieldId);
```

Comment:
So, interesting thing about this - we could mirror what the schema bundle does and only store field ids per type and then have the component ids point to the types. Would reduce the amount of space required and simplify this code a tiny bit, I think.

Reply:
Did that now